x-pack/plugins/timelines/server/search_strategy/timeline/index.ts

```diff
@@ -5,48 +5,166 @@
  * 2.0.
  */
 
-import { map, mergeMap } from 'rxjs/operators';
+import { OWNER, RULE_ID, SPACE_IDS } from '@kbn/rule-data-utils/target/technical_field_names';
+import { map, mergeMap, catchError } from 'rxjs/operators';
+import { from } from 'rxjs';
+import {
+  isValidFeatureId,
+  mapConsumerToIndexName,
+  ALERTS_CONSUMERS,
+} from '@kbn/rule-data-utils/target/alerts_as_data_rbac';
+
+import {
+  AlertingAuthorizationEntity,
+  AlertingAuthorizationFilterType,
+  PluginStartContract as AlertingPluginStartContract,
+} from '../../../../alerting/server';
 import {
   ISearchStrategy,
   PluginStart,
+  SearchStrategyDependencies,
   shimHitsTotal,
 } from '../../../../../../src/plugins/data/server';
-import { ENHANCED_ES_SEARCH_STRATEGY } from '../../../../../../src/plugins/data/common';
 import {
   TimelineFactoryQueryTypes,
   TimelineStrategyResponseType,
   TimelineStrategyRequestType,
+  EntityType,
 } from '../../../common/search_strategy/timeline';
 import { timelineFactory } from './factory';
 import { TimelineFactory } from './factory/types';
+import {
+  ENHANCED_ES_SEARCH_STRATEGY,
+  ISearchOptions,
+} from '../../../../../../src/plugins/data/common';
 
 export const timelineSearchStrategyProvider = <T extends TimelineFactoryQueryTypes>(
-  data: PluginStart
+  data: PluginStart,
+  alerting: AlertingPluginStartContract
 ): ISearchStrategy<TimelineStrategyRequestType<T>, TimelineStrategyResponseType<T>> => {
+  let isUsingInternal = false;
```

Comment:
I'm trying to understand the `isUsingInternal` flag here... If I'm reading correctly, it's possible for one query to go out that is not using internal, then for a second to come through that is using internal... Then if a cancel request comes through for the first request, won't `isUsingInternal` still be set to `true` and it will attempt to cancel the request with the internal client? I guess I am just not sure this `isUsingInternal` flag is being used correctly since the strategy is only returned once, yet `isUsingInternal` can change from request to request.

Reply:
Spoke offline about this. Removed flag and need to determine best method forward with cancelations + auth. Ticket already created.